lib/helpers.bash

```diff
@@ -224,6 +224,10 @@ _bash-it-update-() {
 
   cd "${BASH_IT}" || return
 
+  DIFF=$(git diff --name-status)
+  [ ${#DIFF} -gt 0 ] && echo -e "Either commit your changes using git commit, or clean them...\n\n$DIFF" && return 1
+  
+
```

Comment:
* I *love* that you captured the output to display in case of error.

* The `${#DIFF}` check seems odd - Don't we want something like `[[ ! -z "${DIFF}" ]]` ?

* I think you only need 1 blank line after this - 2 lines feels like too much space.

Reply:
@davidpfarrell Regarding the length of string (point 2) I have checked on my end if there are no changes the length of the output is 0 chars

Still, if you want I can move on to latter syntax